### ISSUE TITLE
ADR014 reports any section containing the ordinary word "describe" as placeholder text

### ISSUE THREAD

**Issue by toobuntu** [joshrotenberg/mdbook-lint#481]
Environment: `adrs` 0.10.1 (Homebrew), which depends on `mdbook-lint-rulesets` 0.14.4 with the `adr` feature; macOS 26.6 build 25G72, arm64. The rule is unchanged at `v0.15.1`.

`adrs doctor` on a complete MADR ADR reports its Context section as empty or placeholder when it contains the word "describe". Changing one word makes the warning go away.

```
$ cat docs/decisions/0001-example.md
---
number: 1
title: Example
status: accepted
date: 2026-08-04
---

# Example

## Context and Problem Statement

These three skills describe org-wide procedure, so a copy in each repository would drift.

## Decision Outcome

Chosen: sync them from one place.

### Consequences

- Good, because one edit reaches every repository.

$ adrs doctor
warning: [ADR014] Section '## Context and Problem Statement' appears to be empty or
contains only placeholder text [docs/decisions/0001-example.md:3]

Found 0 error(s), 1 warning(s), 0 info(s)
```

Replacing `describe` with `cover`, changing nothing else:

```
$ adrs doctor
No issues found. Your ADR repository is healthy!
```

The cause is in `crates/mdbook-lint-rulesets/src/adr/adr014.rs`. `PLACEHOLDER_REGEX` lists the bare word `describe` in its alternation, and `is_placeholder_content` returns `true` when that regex matches anywhere in the section rather than when the section consists only of placeholder text. Any section using the verb in ordinary prose is therefore classified as a placeholder.

The doc comment above the regex records that the word boundary was added to stop `described` from matching, and `adrs` carries a regression test for that case citing the fix in `mdbook-lint-rulesets` 0.14.3. The boundary solved the inflection but left the base verb, which is common in exactly the prose this rule reads: an ADR context section frequently describes something.

## Proposed fix

Match the template phrase instead of the verb. The MADR template's placeholder reads `{Describe the context and problem statement, e.g., in free form using two to three sentences...}`, so a pattern anchored on `describe the context and problem statement` keeps the original intent and costs no coverage. The same applies to `fill in` and `add content`, which are also ordinary English outside a template.

An alternative that would fix the whole class: require the placeholder match to account for most of the section rather than appearing in it, which is what the emitted message already claims.

## Related: the `...` literal has the same shape

`PLACEHOLDER_LITERALS` contains `"..."` and is matched with `contains`, so an ellipsis anywhere in a section produces the same warning. Reproduced on the same file by adding one sentence:

```
The alternative, a copy per repository, drifts... and drift is the whole problem.
```

```
warning: [ADR014] Section '## Context and Problem Statement' appears to be empty or
contains only placeholder text [docs/decisions/0001-example.md:3]
```

An ellipsis is ordinary punctuation in prose. `[insert` and `<insert`, the other two literals, do not have this problem.

## Note on severity

Both cases are warnings, so `adrs doctor` still exits 0 and nothing is blocked. The cost is that a healthy ADR carries a permanent warning, which trains a reader to skim past the warnings that matter.

---

Drafted with AI assistance. Verified on the machine and versions above: the reproduction and both outputs were run as shown against `adrs` 0.10.1; the cause was read from `adr014.rs` at the released tag `v0.15.1`, not from `main`; the 0.14.3 history was read from the regression-test comment in `adrs` at tag `v0.10.1`, `crates/adrs-core/src/lint.rs`.

**Comment by joshrotenberg** [comment]
Thanks for this. The report was accurate on every point, and the reproduction made it quick to confirm: I reproduced both cases against current `main` before changing anything, and the cause is exactly where you located it.

Fix in #487, following your primary proposal.

The patterns are now split by how ambiguous they are:

- Markers that do not occur in finished prose (`todo`, `tbd`, `xxx`, `lorem ipsum`, `placeholder`, `write here`, `to be determined/decided`) keep matching anywhere, since a section mentioning `TODO` is a stub regardless of what else it contains.
- `describe`, `fill in`, and `add content` are matched in their template form instead of as bare verbs. Rather than enumerate template sentences, the braced MADR guidance shape is matched generically (`{Describe the context and problem statement, e.g., ...}`), with the unbraced sentence also covered for authors who removed the braces but not the text.
- `"..."` is no longer matched anywhere. A section consisting only of punctuation is still a placeholder via a new check, so a lone `...` section stays covered and the existing ellipsis test passes unchanged.

Both your cases are clean afterwards, while an unreplaced template and a bare `...` section still warn.

On your alternative: requiring the match to account for most of the section is the better structural fix, and you are right that it is what the message already claims. I took the narrower change here because it costs no coverage, but I have kept the idea on the record in the PR. It would also cover ambiguous uses of `placeholder` and `to be determined`, which still match anywhere.

You are on `adrs` 0.10.1 against `mdbook-lint-rulesets` 0.14.4, so this will reach you when the next release goes out. v0.16.0 is queued in #484.
